common/src/com/thoughtworks/go/config/materials/SubprocessExecutionContext.java

```diff
@@ -24,4 +24,5 @@
 public interface SubprocessExecutionContext {
     String getProcessNamespace(String fingerprint);
     Map<String, String> getDefaultEnvironmentVariables();
+    boolean bare();
```

Comment:
I think we let this slip in the last commit by adding the ```isGItShallowClone``` but ```SubprocessExecutionContext ``` is something used by all materials. We must not pollute it with git related properties. 

For the bare implementation i think we must check if the execution context is on the server and checkout a bare repository on the git material instead of having this specific flag.

And this is a forward only change, the git log commands must work with non bare repositories, that way we don't have to mess with migrations.